# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1549326566 28800
#      Mon Feb 04 16:29:26 2019 -0800
# Node ID 2b6aa96a5c8249b2bd4ceecd158fd280958c8822
# Parent  33f79d94f21f8d1d8b6dcfd1420c952f2e3f77b0
create BIST temporary files in tempdir instead of P_tmpdir (bug #55565)

* help.cc: Use tempdir instead of P_tmpdir in addpath and tempname.
* variables.cc: Use tempdir instead of P_tmpdir to construct the full file
name of the temporary file.

diff --git a/libinterp/corefcn/help.cc b/libinterp/corefcn/help.cc
--- a/libinterp/corefcn/help.cc
+++ b/libinterp/corefcn/help.cc
@@ -836,22 +836,22 @@ the return value is an empty cell array.
         retval(i++) = octave_value (new octave_fcn_handle (nm_fcn->second, nm));
     }
 
   return ovl (retval);
 }
 
 /*
 %!test
-%! f = tempname (P_tmpdir, "oct_");
+%! f = tempname (tempdir (), "oct_");
 %! [~, fcn_name] = fileparts (f);
 %! f = [f ".m"];
 %! save_path = path ();
 %! unwind_protect
-%!   addpath (P_tmpdir);
+%!   addpath (tempdir ());
 %!   fid = fopen (f, "w+");
 %!   fprintf (fid, "function z = %s\n z = localfunctions; end\n", fcn_name);
 %!   fprintf (fid, "function z = b(x)\n z = x+1; end\n");
 %!   fprintf (fid, "function z = c(x)\n z = 2*x; end\n");
 %!   fclose (fid);
 %!   d = eval (fcn_name);
 %!   assert (size (d), [2, 1]);
 %!   assert (d{1} (3), 4);
diff --git a/libinterp/corefcn/variables.cc b/libinterp/corefcn/variables.cc
--- a/libinterp/corefcn/variables.cc
+++ b/libinterp/corefcn/variables.cc
@@ -473,17 +473,17 @@ Octave trusts .oct/.mex files instead of
 ## Don't search path for absolute filenames
 %!test
 %! tname = tempname ();
 %! unwind_protect
 %!   ## open/close file to create it, equivalent of touch
 %!   fid = fopen (tname, "w");
 %!   fclose (fid);
 %!   [~, fname] = fileparts (tname);
-%!   assert (exist (fullfile (P_tmpdir (), fname), "file"), 2);
+%!   assert (exist (fullfile (tempdir (), fname), "file"), 2);
 %! unwind_protect_cleanup
 %!   unlink (tname);
 %! end_unwind_protect
 %! assert (exist (fullfile (pwd (), "%nonexistentfile%"), "file"), 0);
 
 %!testif HAVE_CHOLMOD
 %! assert (exist ("chol"), 3);
 %! assert (exist ("chol.oct"), 3);
